lib/src/cli.rs

```diff
@@ -384,17 +384,31 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
         crate::status::get_status_require_booted(sysroot)?;
     let imgref = host.spec.image.as_ref();
     // If there's no specified image, let's be nice and check if the booted system is using rpm-ostree
-    if imgref.is_none()
-        && host
+    if imgref.is_none() {
+        if host
             .status
             .booted
             .as_ref()
             .map_or(false, |b| b.incompatible)
-    {
-        return Err(anyhow::anyhow!(
-            "Booted deployment contains local rpm-ostree modifications; cannot upgrade via bootc. You can run `rpm-ostree reset` to undo the modifications."
-        ));
+        {
+            return Err(anyhow::anyhow!(
+                "Booted deployment contains local rpm-ostree modifications; cannot upgrade via bootc. You can run `rpm-ostree reset` to undo the modifications."
+            ));
+        }
+
+        if host.status.staged.is_some()
```

Comment:
We should be able to drop this part; I know Rust combinators can be quite obscure but it's basically what the `map_or(false)` is doing here...if the option is None, then the result will be `false`.

Reply:
:+1: thanks for the tip